File: scripts/build_wasm.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'fileutils'

bin_dir = ENV['BIN_DIR'] || 'dist'
FileUtils.mkdir_p(bin_dir)

puts 'Building WASM via ruby-wasm. It requires ruby.wasm packager.'

unless system("rbwasm build -o #{File.join(bin_dir, 'ruby.wasm')} --disable-gems")
  puts 'Failed to build ruby.wasm'
  exit 1
end

unless system("rbwasm pack #{File.join(bin_dir, 'ruby.wasm')} --dir src::/src --dir bin::/bin -o #{File.join(bin_dir, 'cdd-ruby.wasm')}")
  puts 'Failed to pack wasm'
  exit 1
end

File: scripts/build_wasm.py
#!/usr/bin/env python3
import os
import subprocess
import sys

bin_dir = os.environ.get('BIN_DIR', 'dist')
os.makedirs(bin_dir, exist_ok=True)

print('Building WASM via ruby-wasm. It requires ruby.wasm packager.')

ruby_wasm_path = os.path.join(bin_dir, 'ruby.wasm')
if subprocess.run(['rbwasm', 'build', '-o', ruby_wasm_path, '--disable-gems']).returncode != 0:
    print('Failed to build ruby.wasm')
    sys.exit(1)

cdd_ruby_wasm_path = os.path.join(bin_dir, 'cdd-ruby.wasm')
if subprocess.run(['rbwasm', 'pack', ruby_wasm_path, '--dir', 'src::/src', '--dir', 'bin::/bin', '-o', cdd_ruby_wasm_path]).returncode != 0:
    print('Failed to pack wasm')
    sys.exit(1)
